core/src/main/kotlin/org/evomaster/core/search/service/monitor/InterfaceAdapter.kt

```diff
@@ -0,0 +1,170 @@
+package org.evomaster.core.search.service
+
+import com.google.gson.*
+import io.swagger.util.Json
+import org.evomaster.core.problem.rest.RestCallAction
+import org.evomaster.core.search.Action
+import org.evomaster.core.search.gene.Gene
+import java.lang.reflect.Constructor
+import java.lang.reflect.Parameter
+
+import java.lang.reflect.Type
+import kotlin.reflect.KProperty1
+import kotlin.reflect.KVisibility
+import kotlin.reflect.full.allSuperclasses
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.full.isSubclassOf
+import kotlin.reflect.jvm.javaType
+import kotlin.reflect.jvm.jvmErasure
+
+/**
+ * @author: manzhang
+ * @date: 2018/9/11
+ */
+
+class InterfaceAdapter<T : Any> : JsonSerializer<T>, JsonDeserializer<T> {
+
+
+
+    companion object {
+        private val CLASSNAME = "CLASSNAME"
+        private val DATA = "DATA"
+
+        private var fake : JsonObject? = null
+
+        private fun getFakeJson() : JsonObject{
+            var obj = JsonObject()
+            obj.addProperty("maxForRandomizantion", "16")
+            obj.addProperty("value", "fake")
+            obj.addProperty("minLength", "0")
+            obj.addProperty("maxLength", "16")
+            obj.addProperty("name", "fakeArrayGene")
+            return obj
+        }
+    }
+
+    init {
+        fake = getFakeJson()
+    }
+
+    @Throws(JsonParseException::class)
+    override fun deserialize(jsonElement: JsonElement, type: Type, jsonDeserializationContext: JsonDeserializationContext): T {
+        val jsonObject = jsonElement.asJsonObject
+
+        //FIXME MAN: temp solution for Action
+        if(type.typeName.equals(Action::class.java.name) ){
+            return jsonDeserializationContext.deserialize(jsonObject, RestCallAction::class.java)
+        }
+        val prim = jsonObject.get(CLASSNAME) as JsonPrimitive
+        val className = prim.asString
+        val klass = getObjectClass(className)
+
+        // FIXME MAN : the solution should be not hardcode as "gene" attribute.
+        var data =jsonObject.get(DATA).asJsonObject
+        if(containGenProperty(klass)){
+            return deserializeGenProperty(data, klass, jsonDeserializationContext) as T
+        }
+        return jsonDeserializationContext.deserialize(data, klass)
+    }
+
+    override fun serialize(jsonElement: T, type: Type, jsonSerializationContext: JsonSerializationContext): JsonElement {
+        val jsonObject = JsonObject()
+        jsonObject.addProperty(CLASSNAME, jsonElement.javaClass.name)
+        var data = jsonSerializationContext.serialize(jsonElement)
+
+        if(jsonElement is Gene){
+            // FIXME MAN : the solution should be not hardcode as "gene" attribute.
+            // For some reasons, some genes are generic type, but some of them are not, such as SQL.
+            var dprops = jsonElement::class.declaredMemberProperties.filter {  f-> f.name.equals("gene") }
```

Comment:
instead of "f -> f", you can use directly "it"

Reply:
modified it accordingly.